sapi/cli/php_cli_server.c

```diff
@@ -1385,11 +1385,12 @@ static void php_cli_server_request_translate_vpath(php_cli_server_request *reque
 	if (request->vpath_len > 0 && request->vpath[0] != '/') {
 		*p++ = DEFAULT_SLASH;
 	}
-	q = request->vpath + request->vpath_len;
-	while (q > request->vpath) {
-		if (*q-- == '.') {
-			is_static_file = 1;
-			break;
+	q = request->vpath;
```

Comment:
This part looks not quite correct. Fe how would be `/hello.php/world.php` handled? Iterating backwards seems to make more sense. Also some backward compatibility might be broken.

Reply:
The code before my patch, would raise a 404 unless a file corresponding to the path `/hello.php/world.php` is not present, because the path contains a dot.

We can choose to make the recursive search for a router script for ANY file that does not exists, and I agree with this idea. I was only trying to be the less disruptive as possible with the previous behaviour.

For what concerns the backward compatibility: the built-in server is only used for development purposes and must not be used in production.

This new implementation serves all the files that the old one did, but it serves also other paths that instead would have got a 404 status.

As an example, Drupal 8 now works, while before it did not.